### DISCUSSION TITLE
Nested components

### DISCUSSION THREAD

**Discussion by skarensmoll** [radix-ui/primitives#2204] [Help] [answered]
Hi Radix team,

I have a question regarding the impact of nesting components because of the way radix works. Some team members have expressed concerns about the performance implications of highly nested components. For example, using a dropdown with multiple tooltips inside and potentially leading to performance issues.

Considering this, I'm curious about your perspective on such scenarios. Could highly nested components actually decrease performance? I've considered the possibility of implementing lazy loading and Portals, but I'm interested to hear if you have any strong arguments in favor of the nesting approach.

Thank you!

**Comment by benoitgrelard** [comment, marked as answer]
As far as we know it shouldn't really be a concern.
Consider that when you render these big primitives, most of these parts aren't mounted in the DOM at all.
Unless it's open, any big primitive like Dialog, DropdownMenu, Tooltip, Popover, etc, doesn't even mount the content at all.
This means none of the complex logic like positioning, etc is being run.
You typically don't have more than a handful of things open at once ever really.
